📚 DOC: updating intro docs

choldgraf opened the pull request:
This updates the getting started and overview docs, per some recent conversation in #1019 

demo the new overview page here: https://deploy-preview-1023--jupyter-book.netlify.app/start/overview.html

In particular would be curious what @AllenDowney and @emdupre think about this.

closes #1019  (I think)

AllenDowney commented:
This looks really good.  I think the sequence of topics makes sense, and the links to pages with more detail are useful.

Maybe call the page "Installation and overview", since that's the order.  Or split into two pages?

The discussion of the cookie-cutter option is a little confusing to me. The title suggests that the difference is simple versus complex.  But the text makes it sound like cookie-cutter is more advanced (in what way?).  And it seems like the real difference is not complexity but a different use case: if you already have a repo, you might use create; if not, you might use cookie-cutter.

Also, to me "cookie-cutter" means making several identical copies, so I'm not sure it's the right metaphor.

I like that the example does not involve sections.  I suspect that many common use cases will not involve sections.  In my initial explorations, I got pulled into a black hole trying to figure out how sections work, and now it turns out I never needed them in the first place.

choldgraf commented:
Hmmm we got (I think?) a jupyter-cache bug with this build as well?

```
WARNING: Execution Failed: /Users/runner/work/jupyter-book/jupyter-book/docs/file-types/jupytext.Rmd
WARNING: Couldn't find cache key for notebook file docs/file-types/jupytext.Rmd. Outputs will not be inserted.
```

https://github.com/executablebooks/jupyter-book/pull/1023/checks?check_run_id=1216809869#step:6:85

I'm not sure why this error would pop up. Maybe @chrisjsewell or @AakashGfude knows?

choldgraf commented:
thanks for this feedback @AllenDowney ! I just pushed another commit, let's see if that improves things.

I moved the installation section to the landing page and renamed the old one to just `Overview`.

Demo here: https://deploy-preview-1023--jupyter-book.netlify.app/start/overview.html

AllenDowney commented:
@choldgraf  I think this is looking better and better.  I like that the overview is short and each section shows a common case, with links to more details.

emdupre commented on docs/start/overview.md:
````diff
@@ -85,138 +130,124 @@ And it will generate a TOC for you. Note that there must be at least one content
+```yaml
+# In _toc.yml
+- file: landing-page
+- file: page1.ipynb
````
Since the next line states that there should be no extension, should we remove the `.ipynb` here ?

choldgraf replied:
ah definitely, will update soon

emdupre reviewed (commented):
I think this looks good !! I have one question on content, but I think this structure is really helpful :+1:

emdupre commented on docs/start/overview.md:
````diff
@@ -85,138 +130,124 @@ And it will generate a TOC for you. Note that there must be at least one content
+```yaml
+# In _toc.yml
+- file: landing-page
+- file: page1.ipynb
````
```suggestion
- file: page1
```

choldgraf replied:
ah I think I just spot-checked this one in the latest push :-) sorry for crossing updates!

choldgraf commented:
OK so tests are passing and I've incorporated feedback from you both, thanks very much for giving it :-)

I'm sure there is still much to improve (and I'm happy to continue iterating in issues!) but I wanna merge this in so that the docs are up-to-date in case newcomers show up due to JupyterCon 👍
